streamer/src/nonblocking/quic.rs

```diff
@@ -138,70 +154,102 @@ async fn setup_connection(
 
         let remote_addr = connection.remote_address();
 
-        let (mut connection_table_l, stake) = {
-            const PRUNE_TABLE_TO_PERCENTAGE: u8 = 90;
-            let max_percentage_full = Percentage::from(PRUNE_TABLE_TO_PERCENTAGE);
-
+        let table_and_stake = {
             let staked_nodes = staked_nodes.read().unwrap();
             if let Some(stake) = staked_nodes.stake_map.get(&remote_addr.ip()) {
                 let stake = *stake;
-                let total_stake = staked_nodes.total_stake;
                 drop(staked_nodes);
+
                 let mut connection_table_l = staked_connection_table.lock().unwrap();
                 if connection_table_l.total_size >= max_staked_connections {
-                    let max_connections = max_percentage_full.apply_to(max_staked_connections);
-                    let num_pruned = connection_table_l.prune_oldest(max_connections);
-                    stats.num_evictions.fetch_add(num_pruned, Ordering::Relaxed);
+                    let num_pruned = connection_table_l.prune_random(stake);
```

Comment:
nit: so, `prune_oldest()` won't be called ever for `staked_connection_table`. i think `ConnectionEntry::last_update` can be skipped for that case by trait or enum